src/libraries/Microsoft.Extensions.Http/src/DefaultHttpClientFactory.cs

```diff
@@ -208,7 +215,7 @@ internal void ExpiryTimer_Tick(object? state)
             var expired = new ExpiredHandlerTrackingEntry(active);
             _expiredHandlers.Enqueue(expired);
 
-            Log.HandlerExpired(_logger.Value, active.Name, active.Lifetime);
+            Log.HandlerExpired(_logger, active.Name, active.Lifetime);
```

Comment:
Not handler, but the factory... So the story is:

When the application is shutting down, ServiceProvider gets disposed. (It will also dispose all IDisposable singletons alongside).

Now, the Logger has lazy initialization (since https://github.com/dotnet/runtime/issues/89032), that would happen on the first access to it. The first access happens when _(upd.: this is when I realize I've been stopping the wrong timer - it should have been expiryTimer, not cleanupTimer 🤦‍♀️)_ `ExpiryTimer_Tick` is being called the first time, meaning the first of created handlers has just expired (`ExpiryTimer_Tick` is called from a timer [attached to the cached](https://github.com/dotnet/runtime/blob/1d09abfc4cd252777921241055e8089984c2baf5/src/libraries/Microsoft.Extensions.Http/src/ActiveHandlerTrackingEntry.cs#L71) `HttpMessageHandler` instance).

So if the application is shutting down around the handler expiry, what happened was that:
1. The ServiceProvider gets disposed by the app
2. The callback still fires, because no one has stopped it
3. It accesses Lazy, which accesses ServiceProvider to get ILoggerFactory from it
4. ServiceProvider throws ObjectDisposedException

-----

Now re me realizing it was the wrong timer. Making the factory disposable theoretically would still make the behavior a little bit better, because if _dispoised is set already, ExpiryTimer_Tick will just return. But the proper fix would be to go over all cached handlers and stop their respective timers.

Given the time left before ZBB, I am considering to actually remove all of the changes and _leave only swallowing exceptions_ on ILogger creation. All the timers would still fire, but they were firing before.

Now, I _can_ just only ever swallow ObjectDisposedException, that would fix the reported issue. But if we ever stumble upon some other ILogger creation exception, this will become a regression, since it would crash the process where it was not crashing before.

Thoughts @stephentoub ?